src/iexamine.cpp

```diff
@@ -2223,6 +2223,48 @@ void iexamine::tree_pine(player *p, map *m, const tripoint &examp)
     m->ter_set(examp, t_tree_deadpine);
 }
 
+void iexamine::tree_hickory(player *p, map *m, const tripoint &examp)
+{    
+    std::vector<std::string> menu_items;
+    enum options {
+        HARVEST_NUTS,
+        DIG_ROOTS,
+        CANCEL,
+    };
+    uimenu selectmenu;
+    selectmenu.addentry( HARVEST_NUTS, true, MENU_AUTOASSIGN, _("Harvest nuts.") );
+    selectmenu.addentry( DIG_ROOTS, true, MENU_AUTOASSIGN, _("Dig up roots.") );
+    selectmenu.addentry( CANCEL, true, MENU_AUTOASSIGN, _("Cancel") );
+
+    selectmenu.return_invalid = true;
+    selectmenu.text = _("Select an action");
+    selectmenu.selected = 0;
+    selectmenu.query();
+
+    switch( static_cast<options>( selectmenu.ret ) ) {
+    case HARVEST_NUTS:
+        harvest_tree_shrub(p,m,examp);
+        return;
+    
+    case DIG_ROOTS:
+        if( !p->has_items_with_quality( "DIG", 1, 1 ) ) {
+            add_msg(m_info, _("You have no tool to dig with..."));
+            return;
+        }
+        if(!query_yn(_("Dig up %s? This kills the tree!"), m->tername(examp).c_str())) {
+            return;
+        }
+        m->spawn_item(p->pos(), "hickory_root", rng(1,4) );
+        m->ter_set(examp, t_tree_hickory_dead);
+        p->moves -= 2000 / p->skillLevel("survival") + 100;
```

Comment:
Whoa, I see and a division by zero right there. You're not supposed to do that, don't you know what happens to the universe when you divide by zero?